### PR TITLE
feat(datasource/cdnjs): Enable data validation using io-ts library

### PR DESCRIPTION
**Pull request by zharinov**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
I'm hoping for a solution where we define a single source of truth for types/validation and don't "double" define it - once in TS and once using the library's custom code

**Comment by zharinov** [comment]
This is impossible until we have global strict null checks. For now, any such library would generate type with all fields as nullable (even those defined as strict)

**Comment by zharinov** [comment]
Though, during some transition period we may leverage already defined types and check their compatibility with jlibrary-defined ones.

**Comment by rarkins** [comment]
Maybe it's not worth using such a library until we are ready to get the most benefit from it?

**Comment by viceice** [comment]
> Though, during some transition period we may leverage already defined types and check their compatibility with library-defined ones.

Sounds good to me.

**Comment by zharinov** [comment]
Speaking of benefits, runtime input checks are sufficient enough to pursue on its own IMO. This is orthogonal to our type refactoring, however both activities should meet at some point.

File: lib/datasource/cdnjs/common.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+export type IResponse = CdnjsResponse;
+
+export function validResponse(input: unknown): IResponse {
+  const res: IResponse | null = pipe(
+    Response.decode(input),
+    fold(() => null, identity)
+  );
+  if (!res) {
+    throw new Error('Cdnjs: invalid response');
```
Comment (zharinov) [inline lib/datasource/cdnjs/common.ts:30]: `io-ts` is well-crafted functional library, but we are likely to neglect this purity like I did here in favor of convenience

**Comment by rarkins** [comment]
I think Zod wins this round, so closing io-ts PR

**Pull request closed without merging**